Source/engine/render/scrollrt.cpp

```diff
@@ -632,9 +632,8 @@ void DrawItem(const Surface &out, int8_t itemIndex, Point targetBufferPosition)
 void DrawMonsterHelper(const Surface &out, Point tilePosition, Point targetBufferPosition)
 {
 	int mi = dMonster[tilePosition.x][tilePosition.y];
+
 	mi = std::abs(mi) - 1;
-	if (mi < 0) // negative monster (moving into tile)
-		return;

```

Comment:
... wait you put it under abs() :eyes: That obviously wouldn't work.

Reply:
Also, you can't just remove this cause you combined it with the one from town that takes care of not rendering cows twice.

https://github.com/diasurgical/devilutionX/commit/6d5bbe67826f393ce212428e6b25c58a6999d24f#diff-d22929c836495fab0ca461ae9ba210c84323bec4e6466721791013b6317d4867L639

Probably go check town and see if it either looks wrong or crashes.